src/node.h

```diff
@@ -416,7 +416,11 @@ enum Flags : uint64_t {
   kNoRegisterESMLoader = 1 << 3,
   // Set this flag to make Node.js track "raw" file descriptors, i.e. managed
   // by fs.open() and fs.close(), and close them during FreeEnvironment().
-  kTrackUnmanagedFds = 1 << 4
+  kTrackUnmanagedFds = 1 << 4,
+  // This flag should be set to prevent InspectorAgent initialization from
+  // within the environment. This is used by embedders who wish to manage the
+  // InspectorAgent themselves.
+  kNoInitializeInspector = 1 << 5
```

Comment:
I know there is a `kNoRegisterESMLoader` before, but I still think we should try to be consistent and try to avoid making options starting with `No` (also IIUC this makes the instances tested in the embedder test skip inspector initialization by default?)

Reply:
Hmm - i don't think it does? One would need to intentionally set something like:

```
      node::EnvironmentFlags::Flags flags = node::EnvironmentFlags::kDefaultFlags |
                       node::EnvironmentFlags::kNoInitializeInspector;

      env = node::CreateEnvironment(isolate_data, context, args, exec_args, flags);
```

to prevent the initialization occurring from the way this is currently set up, no?

Reply:
I flipped the logic but thinking about it further - this one genuinely does make more sense as a negative flag given that now i need to make sure to add it to anything that doesn't use `kDefaultFlags`, and i can't think of a case where you wouldn't want to initialize the inspector outside of an embedder case. Needing to ensure we also add `kInitializeInspector` to any case where we aren't using `kDefaultFlags` feels like it opens up a lot more room for error 🤔

Reply:
I think to keep it as a positive flag, you would need to update the logic here to make the initialization turned on by default?

https://github.com/nodejs/node/blob/09c5942bfd0efe92e0a4f6791681be915752283d/src/env.cc#L343-L347

Also yes, you are right, thinking about it again if you don't update the line above `kNoInitializeInspector` would be 0 as part of the `kDefaultFlags`, so the test should be fine. Although I think it would be nice to add a test case to test the non-default behavior (just to make sure it doesn't crash or anything when someone touches the bootstrap process).